mafiasi/gprot/models.py

```diff
@@ -6,17 +6,38 @@
 
 from mafiasi.teaching.models import Course, Teacher
 
+
 def make_filename(obj, prefix, ext):
     return '{0}/{1}-{2}.{3}'.format(
         prefix, hexlify(os.urandom(16)), obj.pk, ext)
 
+
 def make_gprot_filename(gprot, filename):
     return make_filename(gprot, 'gprot', 'pdf')
 
+
 def make_attachment_filename(attachment, filename):
     ext = attachment.mime_type.split('/')[-1]
     return make_filename(attachment, 'gprot-attachment', ext)
 
+
+class LabelManager(models.Manager):
+
+    def as_choices(self):
+        for label in self.all().order_by('name'):
+            yield label.pk, label.name
+
+
+class Label(models.Model):
+    name = models.CharField(max_length=50)
+    color = models.CharField(max_length=15, default="default")
```

Comment:
I think we should set `choices` to the color field to simplify the selection in the admin interface